Assets/MixedRealityToolkit/_Core/Definitions/Devices/MixedRealityInteractionMapping.cs

```diff
@@ -0,0 +1,120 @@
+    [Serializable]
+    public class MixedRealityInteractionMapping<TReadingType> : IMixedRealityInteractionMapping
+    {
+        /// <summary>
+        /// The constructor for a new Interaction Mapping definition
+        /// </summary>
+        /// <param name="id">Identity for mapping</param>
+        /// <param name="axisType">The axis that the mapping operates on, also denotes the data type for the mapping</param>
+        /// <param name="inputType">The physical input device / control</param>
+        /// <param name="inputAction">The logical InputAction that this input performs</param>
+        public MixedRealityInteractionMapping(uint id, AxisType axisType, DeviceInputType inputType, IMixedRealityInputAction inputAction)
+        {
+            this.id = id;
+            this.axisType = axisType;
+            this.inputType = inputType;
+            this.inputAction = inputAction;
+            changed = false;
+        }
+
+        #region Interaction Properties
+
+        [SerializeField]
+        private uint id;
+
+        /// <inheritdoc/>
+        public uint Id => id;
+
+        [SerializeField]
+        [Tooltip("The axis type of the button, e.g. Analogue, Digital, etc.")]
+        private AxisType axisType;
+
+        /// <inheritdoc/>
+        public AxisType AxisType => axisType;
+
+        [SerializeField]
+        [Tooltip("The primary action of the input as defined by the controller SDK.")]
+        private DeviceInputType inputType;
+
+        /// <inheritdoc/>
+        public DeviceInputType InputType => inputType;
+
+        [SerializeField]
+        [Tooltip("Action to be raised to the Input Manager when the input data has changed.")]
+        private IMixedRealityInputAction inputAction;
+
+        /// <inheritdoc/>
+        public IMixedRealityInputAction InputAction => inputAction;
+
+        private TReadingType currentReading;
+
+        private bool changed;
+
+        /// <inheritdoc/>
+        public bool Changed
+        {
+            get
+            {
+                bool returnValue = changed;
+
+                if (changed)
+                {
+                    changed = false;
+                }
+
+                return returnValue;
+            }
+            private set
+            {
+                changed = value;
+            }
+        }
+
+        #endregion Interaction Properties
+
+        #region Operators
+
+        /// <summary>
+        /// Get the value of the Interaction Mapping data
+        /// </summary>
+        /// <returns>The current reading for the mapping</returns>
+        public TReadingType GetValue() => currentReading;
+
+        /// <summary>
+        /// Get the value of the Interaction Mapping data
+        /// </summary>
+        /// <param name="reading">Output parameter for the reading value, returns the current reading for the mapping</param>
+        public void GetValue(out TReadingType reading)
+        {
+            reading = currentReading;
```

Comment:
I'd like to keep the direct calls to do a speed test.

Reply:
Then following this PR, create some unit tests to compare.  I compared before and it was comparable but easier to manage